### PR TITLE
Feat: streams

### PR DESCRIPTION
**Pull request by PaulDremanovich**
Task: [Initial version of IPFS node](https://trello.com/c/d1iMesi5/386-initial-version-of-ipfs-node)

Completed items:
- Modify `POST /file/upload` as it can upload files using streams or use other storage than [multer.MemoryStorage](https://expressjs.com/en/resources/middleware/multer.html). Currently uploading large files may cause running out of memory. Also add `maxFileSize` to the config.
- Rewrite `GET /file/:cid` using streams
- Move the request timeout of `GET /file/:cid` to the config file.

### CODE REVIEW

**Comment by PaulDremanovich** [comment]
[Initial version of IPFS node](https://trello.com/c/d1iMesi5/386-initial-version-of-ipfs-node)

File: config.default.json5 [hunk 1]
```diff
@@ -31,5 +31,7 @@
     ]
   },
   serverPort: 4000,
-  autoPeeringPeriod: '*/10 * * * * *'
+  autoPeeringPeriod: '*/10 * * * * *',
+  uploadLimitSizeBytes: 268435456,
+  findFileTimeout: 30000
```
Comment (bludnic) [inline config.default.json5]: add comment in MB

File: src/index.ts [hunk 2]
```diff
@@ -334,54 +331,61 @@ app.get('/pins/isPinned/:cid', async (req, res) => {
...
+        error: 'No file uploaded'
+      })
+    }
+    const files = flatFiles(req.files)
+    pino.logger.info(`req.files: : ${JSON.stringify(files.map((item) => item.originalName))}`)
 
-    const { buffer, originalname } = file
+    const cids: CID[] = []
+    for (const file of files) {
+      console.log(`Adding ${file.originalName} to IPFS`)
```
Comment (bludnic) [inline src/index.ts]: use pino logger

File: src/multer.d.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      size: number
+      stream: ReadStream
+      detectedMimeType: string
+      detectedFileExtension: string
+      clientReportedMimeType: string
+      clientReportedFileExtension: string
+    }
+  }
+  export default function multer(options: {}): Multer
+}
```
Comment (bludnic) [inline src/multer.d.ts]: Just install `@types/multer`. You don't need this file.
https://www.npmjs.com/package/@types/multer

```
npm i --save-dev @types/multer
```

**Review by bludnic — CHANGES_REQUESTED** [review]

File: src/multer.d.ts (hunk 3 above)
Comment (PaulDremanovich) [reply src/multer.d.ts]: There are no types yet for Multer version 2.0.0-rс4

Comment (bludnic) [reply src/multer.d.ts]: Can you use `1.4.5-lts.1`? Or you need `2.0.0-rс4`?

image

File: config.default.json5 [hunk 4]
```diff
@@ -32,6 +32,6 @@
   },
   serverPort: 4000,
   autoPeeringPeriod: '*/10 * * * * *',
-  uploadLimitSizeBytes: 268435456,
+  uploadLimitSizeBytes: 268435456, //256 Mb.
```
Comment (bludnic) [inline config.default.json5]: add space -> `// 256 Mb.`

Comment (bludnic) [reply config.default.json5]: Check the linter/prettier. It should format this automatically, I think.

**Review by RealGoodProgrammer — APPROVED** [review]

**Review by bludnic — APPROVED** [review]

**Pull request merged**